google/cloud/config.cmake.in

```diff
@@ -15,5 +15,6 @@
 include(CMakeFindDependencyMacro)
 find_dependency(Threads)
 find_dependency(absl)
+@GOOGLE_CLOUD_CPP_FIND_OPTIONAL_DEPENDENCIES@
```

Comment:
We need something similar for the generated `.pc` file.

Reply:
I don't think opentelemetry installs a `.pc` file. https://github.com/open-telemetry/opentelemetry-cpp/blob/57bf8c2b0e85215a61602f559522d08caa4d2fb8/CMakeLists.txt#L512-L544. Maybe because the API is self contained?

These are the artifacts I see:
```
docker:fedora-36-cmake$ ls /usr/local/lib64/pkgconfig/ | grep telemetry

docker:fedora-36-cmake$ ls /usr/local/lib64/cmake/ | grep telemetry
opentelemetry-cpp

docker:fedora-36-cmake$ ls /usr/local/lib64/ | grep telemetry
libopentelemetry_common.so
libopentelemetry_exporter_in_memory.so
libopentelemetry_exporter_ostream_metrics.so
libopentelemetry_exporter_ostream_span.so
libopentelemetry_http_client_curl.so
libopentelemetry_metrics.so
libopentelemetry_resources.so
libopentelemetry_trace.so
libopentelemetry_version.so
```

Note that the libs come from the SDK: https://github.com/open-telemetry/opentelemetry-cpp/tree/main/sdk/src. If we build with `-DWITH_API_ONLY` they go away.

Reply:
Even if the API is self-contained, the headers may be needed (e.g. if they were installed in a non-standard location).

This might be a problem.  But we can punt if otel does not provide support for pkg-config.

Reply:
https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#incorporating-into-an-existing-cmake-project

ah, it looks like I am supposed to be using this:
```
target_include_directories(foo PRIVATE ${OPENTELEMETRY_CPP_INCLUDE_DIRS})
target_link_libraries(foo PRIVATE ${OPENTELEMETRY_CPP_LIBRARIES})
```

Reply:
That sounds plain wrong.  Not your fault, their advice is plain wrong.  That is not how modern CMake libraries are supposed to be consumed.  One should not need to reference `${BLAH_BLAH_INCLUDE_DIRS}` directly.  The include directories should be part of the `blah blah target`.

Also, it does not fix the problem.  We need to figure out how to use these in plain `make(1)`, or anything else that consumes `pkg-config` output.

Reply:
Hm.. I will merge the PR as is.

Reply:
I opened https://github.com/open-telemetry/opentelemetry-cpp/issues/1850